docs/INSTALLATION.md

```diff
@@ -1,65 +1,82 @@
-have installed it correctly by running this command in a terminal:
+## Get GHCup
 
-```bash
-stack --version
-```
+The recommended way of installing Haskell tools is through [GHCup][ghcup].
 
-If it outputs a Stack version equal to or greater than 1.1.2, then you have it
-installed correctly.
+With GHCup you can manage your installed compilers, project/package managers, and language server.
 
-## Installing GHC
 
-Stack will automatically download and install GHC (the Glasgow Haskell
-compiler) when it is required. You can start the installation and test
-by running the following command in a terminal:
+## Install Stack
 
-```bash
-stack ghc
-```
+For doing Exercism exercises you only need Stack.
+
+To install Stack, simply run the GHCup installation command or script and follow the defaults but specify that you would like to have Stack installed as well.
 
-If this outputs something like `ghc: no input files`, GHC is installed
-correctly.
+Did you miss the opportunity? No worries: just run `ghcup install stack` afterwards.
 
-Note that there are other ways to install GHC on a system (like the
-[Haskell platform](https://www.haskell.org/platform/)), but for
-running the Exercism tests, a regular stack installation is most
-likely to work correctly.
+GHCup will install
 
-## Installing HLint (optional)
+- **A compiler** (GHC: the Glasgow Haskell Compiler)
+  
+  Strictly speaking you do not need this for Exercism, but it comes in handy when you want to play with GHCi outside of exercise directories.
 
-HLint is a tool for suggesting possible improvements to Haskell code. These
-suggestions include ideas such as using alternative functions, simplifying
-code and spotting redundancies.
+- **Cabal**
 
-You can use Stack to install HLint:
+  A project/package manager. You do not need this for Exercism.
+
+- **Stack**
+
+  A project/package manager that you do need for Exercism. It will make sure you have the right compiler and libraries for each individual project (exercise).
+  
+To check that installation was successful, run the command `ghcup list` and/or `stack --version`.
+
+
+## Get the right GHC version
+
+Stack will automatically download and install GHC when it is required. You can start the installation and test
+by running the following command in a terminal:
 
 ```bash
-stack install hlint
+# move into exercise directory
+cd /exercism/haskell/hello-world
+# start interactive environment
+stack ghc -- --version
 ```
 
-Check that it was installed correctly with:
+If this outputs something like
 
-```bash
-hlint --version
 ```
+The Glorious Glasgow Haskell Compilation System, version 8.8.4
+```
+
+then GHC is installed correctly.
+
+
+## Set up your editor to work with Haskell
+
+Haskell support is provided by the Haskell Language Server (HLS).
+
+
+### Visual Studio Code
+
+Just get [this extension][vscode-haskell]. It will use GHCup to install HLS automatically, after which you'll be all set.
```

Comment:
```suggestion
Just get the [Haskell for Visual Studio Code extension][vscode-haskell]. It will use GHCup to install HLS automatically, after which you'll be all set.
```

By my reading, a link like "this extension" is like "click here", which I was taught by https://webaccess.berkeley.edu/ask-pecan/click-here to avoid, so I have decided to suggest this.